feat (catalog): Support uploading OSCAL files

mpemy opened the pull request:
Closes https://github.com/EasyDynamics/oscal-react-library/issues/888

Bronstrom commented on packages/oscal-react-library/src/components/OSCALCatalogBaseline.tsx:
```diff
@@ -466,7 +420,11 @@ export default function OSCALCatalogBaseline() {
   address = authorAddress;
   const createdModel = newOSCALModel === undefined ? Data : newOSCALModel;
-  const LabelText = openCatalogBaseline ? createdModel.title : " Catalogs & Baselines";
+  const LabelText = openCatalogBaseline
+    ? createdModel.title
+    : uploadNewCatalogBaseline
+    ? " Upload a Catalog or Baseline "
+    : " Catalogs & Baselines";
```
I'm having a little difficulties reading this. To keep the code clear and concise, I would suggest breaking this into two constants, each handling one of the ternary operations.

mpemy replied:
@Bronstrom Thanks for the insight, I broke it down to ternary operations.

Bronstrom commented on packages/oscal-react-library/src/components/OSCALCatalogBaseline.tsx:
```diff
@@ -520,6 +478,472 @@ export default function OSCALCatalogBaseline() {
+        {upload && (
+          <Box
+            sx={{
+              top: 180,
+              left: 40,
+              height: 400,
+              width: 1030,
+              background: "#ffffff",
+              position: "absolute",
+              border: "1xp solid #0028675E",
```
Just wanted to just mention it once more, in the future we should try to keep the units of measure to `rem`, where we are sizing/positioning components.

Bronstrom commented on packages/oscal-react-library/src/components/OSCALCatalogBaseline.tsx:
```diff
@@ -1154,10 +1624,58 @@ export default function OSCALCatalogBaseline() {
+                <option value="CO">Colorado</option>
+                <option value="CT">Connecticut</option>
+                <option value="DE">Delaware</option>
+                <option value="DC">District of Columbia</option>
```
Is this the same `Select` field and options used in #927? I think we could refactor this code and reuse it, once one of these PRs are merged in.

Bronstrom commented on packages/oscal-react-library/src/components/OSCALLoader.tsx:
```diff
@@ -28,6 +29,7 @@ import {
 import { ReactElement } from "react-markdown/lib/react-markdown";
 import { AnchorLinkProps } from "./OSCALAnchorLinkHeader";
 
+const isDevMode = false; // This boolean is used to switch to dev mode
```
Do we need a second `isDevMode` flag in the loader? Is there an issue with just using the one in the App? Maybe it can be placed in a context or be passed down as `props` to the loader.

Bronstrom commented on packages/oscal-react-library/src/components/images/icons/fileUpload.svg:Zone.Identifier:
```diff
@@ -0,0 +1,3 @@
+[ZoneTransfer]
+ZoneId=3
+HostUrl=https://www.figma.com/
```
Same comment here on the `svg`s as #927: we want these to be `svg`s with path code instead of figma links.

Bronstrom commented on packages/oscal-react-library/src/components/OSCALCatalogBaseline.tsx:
```diff
@@ -520,6 +478,472 @@ export default function OSCALCatalogBaseline() {
+            UPLOAD
+          </Typography>
+        </OSCALBreadCrumbs>
+      );
+    }
+    return (
+      <OSCALBreadCrumbs
+        aria-label="breadcrumb"
+        maxItems={3}
+        itemsBeforeCollapse={1}
+        itemsAfterCollapse={2}
+      >
+        {/* <OSCALLink href="/">HOME</OSCALLink> */}
+        <OSCALTertiaryGrayscaleButton>
+          <Typography
+            sx={{
+              ":hover": {
+                color: "#ff6600",
+              },
+            }}
+            onClick={loadPage}
+          >
+            Home
+          </Typography>
+        </OSCALTertiaryGrayscaleButton>
+        <Typography
+          color="primary"
+          sx={{
+            fontWeight: (theme) => theme.typography.fontWeightBold,
+            textTransform: "uppercase",
+            ":hover": {
+              color: "#ff6600",
+            },
+            fontFamily: "Source Sans Pro",
+          }}
+        >
+          CATALOGS & BASELINES
+        </Typography>
+      </OSCALBreadCrumbs>
+    );
```
I'm trying to pick out the differences between these three conditions here, because it looks like a lot of reusable code. Just an idea, but can you just validate these conditions when attempting to render parts of this code? For example, with the first condition it looks like you're trying to figure out whether to include this breadcrumb item:
```
        {(item?.model.title?.length > 0) &&
          <CatalogTooltip title={usedTitle.toUpperCase()}>
            <Typography
              color="primary"
              sx={{
                fontWeight: (theme) => theme.typography.fontWeightBold,
                textTransform: "uppercase",
                ":hover": {
                  color: "#ff6600",
                },
                fontFamily: "Source Sans Pro",
              }}
            >
              {trunckatedTitle.toUpperCase()}
            </Typography>
          </CatalogTooltip>
         }

```

mpemy commented:
@Bronstrom, I addressed code review comments and suggestions. As for the flag isDevMode, I tried to add it as a global constant but the issue it creates is that the package oscal-viewer should be set as a dependency for the other packages. But those packages are already dependencies for oscal-viewer.
I also tried OSCALContext but it did not work. Given that we will soon have a dev branch, we will not have this issue once that branch is established.

Bronstrom reviewed (commented):
Thanks for working hard on the palette updates! One small change before we merge this PR in:

mpemy commented on packages/oscal-react-library/src/components/OSCALCatalogBaseline.tsx:
```diff
@@ -1259,9 +1934,12 @@ export default function OSCALCatalogBaseline() {
+                {/* <AddInformationLabel>
                   Add information about the {Model.toLowerCase()}:
-                </AddInformationLabel>
+                </AddInformationLabel> */}
```
@Bronstrom, Sorry I just forgot to delete that line before checking the code.
I just addressed this issue. Please also review the PR feat (catalog): Manage catalog groups #927. I also made similar changes on styling on Manage Group.

Bronstrom replied:
No worries, appreciate you addressing this quickly!
